ckan/logic/action/delete.py

```diff
@@ -185,6 +184,15 @@ def resource_delete(context: Context, data_dict: DataDict) -> ActionResult.Resou
         plugin.before_resource_delete(context, data_dict,
                                       pkg_dict.get('resources', []))
 
+    # Delete file if it was uploaded
+    if pkg_dict.get('url_type') == 'upload':
```

Comment:
We should really add a feature flag on this to not delete so it could be included in core.